src/Engine/Combat/StrikeParams.java

```diff
@@ -100,6 +109,9 @@ protected StrikeParams(
     this.attackPower = attacker.attackPower;
     this.luckRolled = attacker.CO.luck;
     this.isCounter = isCounter;
+    aw1Luck = attacker.CO.aw1Combat;
+    if( aw1Luck && isCounter )
+      this.attackerHealth = attacker.health;
```

Comment:
This makes it use precise health instead of rounded health for the calculation? Might toss that in a comment. Yes, that's basically what the code says, to the discerning eye, but it's also an odd thing to switch to exact HP just for AW1 and only for counterattacks, and I like having a comment to affirm non-obvious intent.